src/components/body/body-cell.component.ts

```diff
@@ -23,6 +25,46 @@ import { MouseEvent, KeyboardEvent } from '../../events';
           (click)="onCheckboxChange($event)"
         />
       </label>
+      <label class="clickable"
+        *ngIf="column.isTreeColumn"
+        (click)="onTreeAction()">
+        <span *ngIf="_treeStatus==='loading' &&
+                !column.treeLoaderTemplate">
+          [=]
+        </span>
+        <ng-template *ngIf="_treeStatus==='loading' &&
+                column.treeLoaderTemplate"
+          [ngTemplateOutlet]="column.treeLoaderTemplate">
+        </ng-template>
+
+        <span *ngIf="_treeStatus==='collapsed' &&
+                !column.treeExpanderTemplate">
+          [+]
+        </span>
+        <ng-template *ngIf="_treeStatus==='collapsed' &&
+                column.treeExpanderTemplate"
+          [ngTemplateOutlet]="column.treeExpanderTemplate">
+        </ng-template>
+
+        <span *ngIf="_treeStatus==='expanded' &&
+                     !column.treeCollapserTemplate">
+          [-]
+        </span>
+        <ng-template *ngIf="_treeStatus==='expanded' &&
+                            column.treeCollapserTemplate"
+          [ngTemplateOutlet]="column.treeCollapserTemplate">
```

Comment:
I'd like to consolidate these templates into 1 and pass the state in a `let` flag.

Reply:
Thinking about this more, do we really need a specific template for this? Why can't they just use a normal column template and us expose the event they can trigger?

Reply:
The integrator may want to choose their own icon classes. We are exposing the event but not changing the state inside. Ths the integration has full ability to change the state.